parse.y

```diff
@@ -10926,8 +10926,7 @@ cond0(struct parser_params *p, NODE *node, int method_op, const YYLTYPE *loc)
 
       case NODE_DREGX:
 	{
-	    if (!method_op)
```

Comment:
Do not ignore `method_op` flag.
It is `false` when `/re/.!`, and the redefinition is in effect.